### PR TITLE
Removing exception when variable bounds are equal

### PR DESCRIPTION
**Pull request by carldlaird**
PyNumero used to throw an exception when the variable bounds were equal (or crossed). This change only throws an exception if the bounds are crossed.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/contrib/pynumero/interfaces/ampl_nlp.py [hunk 1]
```diff
@@ -129,16 +129,11 @@ def _collect_nlp_structure(self):
...
+        # check to make sure there are no crossed bounds
         bounds_difference = self._primals_ub - self._primals_lb
-        abs_bounds_difference = np.absolute(bounds_difference)
-        fixed_vars = np.any(abs_bounds_difference < tolerance_fixed_bounds)
-        if fixed_vars:
-            print(np.where(abs_bounds_difference<tolerance_fixed_bounds))
-            raise RuntimeError("Variables fixed using bounds is not currently supported.")
+        if np.any(bounds_difference < 0):
+            print(np.where(bounds_difference < 0))
+            raise RuntimeError("Some variables have lower bounds that are larger than the upper bounds.")
```
Comment (michaelbynum) [inline pyomo/contrib/pynumero/interfaces/ampl_nlp.py:136]: is this check also done on the next 2 lines?

**Review by Robbybp — CHANGES_REQUESTED** [review]
Can you add a test that demonstrates that variables with equal upper and lower bounds are now supported?

**Comment by blnicho** [comment]
@carldlaird did you see the comments from Michael and Robby above? Also, please add this to the release board if this is a priority for the release planned for next week.

**Comment by bknueven** [comment]
> Also, please add this to the release board if this is a priority for the release planned for next week.

This PR is needed for our IDAES-derived project [WaterTAP](https://github.com/watertap-org/watertap), so it would be great if it were in the next Pyomo release. If @carldlaird is unavailable I'm happy to open a PR addressing Michael and Robby's concerns.

**Review by jsiirola — APPROVED** [review]

**Review by Robbybp — APPROVED** [review]

**Comment by jsiirola** [comment]
@Robbybp, can we get a test that actually triggers the LB > UB exception?

**Review by blnicho — APPROVED** [review]

**Pull request merged**